### PR TITLE
provide new image size 'original'

### PR DESCRIPTION
**Pull request by der-john**
Adds a new image format called 'original' besides 'detail' and 'thumbnail'.

Tests pass as before, but I haven't written a new one for the new logic yet.

### CODE REVIEW

**Comment by joka** [comment]
Sidenote: we did not do this in the first place to prevent high server due to big image sizes.

**Comment by 2e2a** [comment]
> Sidenote: we did not do this in the first place to prevent high server load due to big image sizes.

The main reason for this pull request ist to keep the original ratio. So we might reduce the the image size.

**Comment by xi** [comment]
> > Sidenote: we did not do this in the first place to prevent high server load due to big image sizes.
>
> The main reason for this pull request ist to keep the original ratio. So we might reduce the the image size.

Absolutely! It's probably best to fix the area, i.e. `width * height == X`.

File: src/adhocracy_core/adhocracy_core/resources/image.py [hunk 1]
```diff
@@ -47,9 +47,12 @@ def get_response(self, registry: Registry=None) -> FileResponse:
         """
         if self._is_resized():
             return self._get_response()
-        elif self.dimensions:
+        elif self.dimensions or self.dimensions is None:
```
Comment (2e2a) [inline src/adhocracy_core/adhocracy_core/resources/image.py]: This is always true. Would need to clearly split the two cases.

File: src/adhocracy_core/adhocracy_core/resources/image.py [hunk 2]
```diff
@@ -65,6 +68,31 @@ def _is_resized(self) -> bool:
...
+                           progressive=True,
+                           quality=80,
+                           optimize=True)
+            else:
+                image.save(bytestream,
+                           image.format)
+            bytestream.seek(0)
+        self.upload(bytestream)
+        self.mimetype = original.mimetype
+
```
Comment (2e2a) [inline src/adhocracy_core/adhocracy_core/resources/image.py]: Avoid code duplication with _upload_crop_and_resize

Comment (2e2a) [inline src/adhocracy_core/adhocracy_core/resources/image.py]: Need to reduce image size. Consider @xi's comment.

File: src/adhocracy_core/adhocracy_core/resources/image.py (hunk 1 above)
Comment (der-john) [inline src/adhocracy_core/adhocracy_core/resources/image.py]: Oops.

**Comment by der-john** [comment]
doesn't seem to be worth the effort anymore, since demo is not being realized.

**Pull request closed without merging**